### ISSUE TITLE
Rebase arc64 glibc port to latest 2.34

### ISSUE THREAD

**Issue by vineetgarc** [foss-for-synopsys-dwc-arc-processors/toolchain#406]
glibc 2.34 was released on 8-1-2021.
For eventual upstreaming (though contingent to gcc/binutils) we need to rebase on latest.

**Comment by vineetgarc** [comment]
The rebase is done and port cleaned up too: current state at [arc64-2.34](https://github.com/foss-for-synopsys-dwc-arc-processors/glibc/tree/arc64-2.34).  It is failing to boot currently.

```
  with environment:
    HOME=/
    TERM=linux
Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00
---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00 ]---

```

**Comment by vineetgarc** [comment]
Currently gated on #407 to be able to debug !

**Comment by vineetgarc** [comment]
Using nSIM mmux instruction tracing was able to sift through the failures to find the rootcase of abort. 
An important function malloc_consolidate() is being stubbed out at the time of compilation due to an abort()

```
malloc_consolidate:
# malloc.c:4632:     p = atomic_exchange_acq (fb, NULL);
	bl	@abort	#
```
That in turn was happening due to a snafu during rebase:
With this rebase I switched glibc atomics to use gcc builtins: much of the old ARCv2 code was shared - which in turn has 64-bit atomics mapped to abort(). For ARCv3/64-bit 64-bit atomics are available.

```diff
diff --git a/sysdeps/arc/atomic-machine.h b/sysdeps/arc/atomic-machine.h

+# if !defined(__ARC64__)
#  define __arch_compare_and_exchange_bool_64_acq(mem, newval, oldval) \
    (abort (), 0)
+#endif
```

**Comment by vineetgarc** [comment]
With a slightly different variant of above fix, subsumed in commit  [ 20e255de52bd ARCv3: baseline ABI](https://github.com/foss-for-synopsys-dwc-arc-processors/glibc/commit/20e255de52bdf10eed62eb58e0163e1c0f1f976d) glibc 2.34 based system boots to shell.
